### PR TITLE
Markdown newlines Fixes #562

### LINKED ISSUE #562: Need unit test for markdown double newline
Need unit test for markdown double newline

### PR DESCRIPTION
**Pull request by reyesrico**
Fixes newlines and <br> tags processing, tests paragraphs created.

### CODE REVIEW

**Comment by billba** [comment]
It seems like this retains the behavior of converting BR tags to newlines? This seems to go against our policy of ignoring all HTML tags.

**Comment by reyesrico** [comment]
@billba  for this particular issue it was fixed that markdown-it can render double lines correctly for the following newline strings:
- \r\n\r\n 
- \n\n
- <br/><br/>

Specifically for the <br> tag, it was only 'updated' the string replacement that it is now living FormatText file in master.
.replace(/<br\s*\/?>/ig, '\n')  

Also, to make sure that these strings are rendered as expected, I developed a test to read the number and content of the paragraphs.

As agreed before, we need to ignore all <br> tags. I suggest to clean this part with another PR.

**Comment by billba** [comment]
Got it. Different bug.

File: test/commands_map.ts [hunk 1]
```diff
@@ -265,6 +265,33 @@ var commands_map: CommandValuesMap = {
             sendActivity(conversationId, server_content.mar_card);
         }
     },
+    "markdown-newlines": {
```
Comment (danmarshall) [inline test/commands_map.ts]: I'd like to see these each broken out into individual tests.

File: test/server_content.ts [hunk 2]
```diff
@@ -341,6 +341,15 @@ export var mar_encode_card: dl.Message = {
     text: "Some links are [already encoded](https://bing.com?q=some%20value) and some have [query strings that need encoding](https://bing.com?q=some value) and these can occur [multiple](https://bing.com?q=some%20value) [times](https://bing.com?q=some value) in one message. (Links can even appear [within](https://bing.com?q=some%20value) [parentheses](https://bing.com?q=some value) too.)"
 }
 
+export var mar_newlines_card: dl.Message = {
+    type: "message",
+    from: bot,
+    timestamp: new Date().toUTCString(),
+    channelId: "webchat",
+    textFormat: "markdown",
+    text: "**Windows** single newline->\r\n<-finish. \r\n\r\n**Unix** single newline->\n<-finish. \r\n\r\n---\r\n\r\n**Windows** double newline->\r\n \r\n<-finish. \r\n\r\n**Unix** double newline->\n \n<-finish. \r\n\r\n---\r\n\r\n**Windows** double-double newline->\r\n \r\n \r\n \r\n<-finish. \r\n\r\n**Unix** double-double newline->\n \n \n \n<-finish. \r\n\r\n---\r\n\r\n**BR** single newline-><br /><-finish. \r\n\r\n **BR** double newline-><br /><br /><-finish. \r\n\r\n**BR** double-double newline-><br /><br /><br /><br /><-finish."
```
Comment (danmarshall) [inline test/server_content.ts]: These should also be their own commands, or a command with a parameter.

File: test/commands_map.ts [hunk 3]
```diff
@@ -265,33 +265,33 @@ var commands_map: CommandValuesMap = {
...
+            var last_bubble = document.querySelector('.wc-message-wrapper:last-child .wc-message.wc-message-from-bot .format-markdown');
+            return last_bubble.getElementsByTagName('p').length === 4 && last_bubble.getElementsByTagName('br').length === 3;
+        },
+        server: function (conversationId, sendActivity) {
+            sendActivity(conversationId, server_content.mar_newlines_double_card);
+        }
+    },
+    "markdown-newlines-double-double": {
+        client: function () {
+            var last_bubble = document.querySelector('.wc-message-wrapper:last-child .wc-message.wc-message-from-bot .format-markdown');
```
Comment (danmarshall) [inline test/commands_map.ts]: very cool!!
Now, can you extract `last_bubble` into a function?

Comment (danmarshall) [reply test/commands_map.ts]: ok, cant do easily :(

**Pull request merged**